### PR TITLE
Start to bring back the 'toit' executable.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/toit.cc [hunk 1]
```diff
@@ -15,375 +15,51 @@
...
-      fprintf(stderr, "Can't have options with '%s'\n", argv[1]);
-      print_usage(1);
-    }
-    print_version();
-  }
-
-  // TODO(2663): remove support for '-r'.
-  if (strcmp(argv[1], "-r") == 0 ||
-      SnapshotBundle::is_bundle_file(argv[1])) {
+  if (argc >= 2 && SnapshotBundle::is_bundle_file(argv[1])) {
```
Comment (kasperl) [inline src/toit.cc:38]: Do we need all this handling?

File: tools/toit.toit [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        cli.Option "sdk-dir"
+            --help="Path to the SDK root."
+            --type="dir"
+            --hidden,
+      ]
+
+  compile-analyze-run-options := [
+    cli.Flag "show-package-warnings"
+        --help="Show warnings from packages.",
+    cli.Flag "Werror" --short-name="Werror"
```
Comment (kasperl) [inline tools/toit.toit:38]: So the short name means you can write `--Werror` and `-Werror`?

File: tools/toit.toit [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        --help="Path to the project root. Any package.lock file must be in that folder."
+        --type="dir",
+    cli.Option "X" --short-name="X"
+        --help="Provide a compiler flag."
+        --hidden
+        --multi,
+  ]
+
+  compile-run-options := [
+    cli.OptionInt "optimization" --short-name="O"
```
Comment (kasperl) [inline tools/toit.toit:50]:
```suggestion
    cli.OptionInt "optimization-level" --short-name="O"
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/toit.cc (hunk 1 above)
Comment (floitsch) [reply src/toit.cc:49]: It's to support fast "run with a snapshot".
For all other cases we need to invoke the compiler anyway, so it feels like running a Toit program first doesn't hurt as much.
We can add more fast cases, if we feel like going through the Toit code adds too much overhead.

File: tools/toit.toit (hunk 2 above)
Comment (floitsch) [reply tools/toit.toit:38]: correct.

File: tools/toit.toit (hunk 3 above)
Comment (floitsch) [reply tools/toit.toit]: done.

**Pull request merged**